src/System.Private.ServiceModel/src/System/IdentityModel/CryptoHelper.cs

```diff
@@ -61,7 +97,305 @@ internal static bool IsAsymmetricAlgorithm(string algorithm)
+        /// <param name="sizeInBits"></param>
+        /// <returns></returns>
+        public static byte[] GenerateRandomBytes(int sizeInBits)
+        {
+            int sizeInBytes = sizeInBits / 8;
+            if (sizeInBits <= 0)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException(nameof(sizeInBits), SR.Format(SR.ID6033, sizeInBits)));
+            }
+            else if (sizeInBytes * 8 != sizeInBits)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentException(SR.Format(SR.ID6002, sizeInBits), nameof(sizeInBits)));
+            }
+
+            byte[] data = new byte[sizeInBytes];
+            GenerateRandomBytes(data);
+
+            return data;
+        }
+
+        internal static RandomNumberGenerator RandomNumberGenerator
+        {
+            get
+            {
+                if (s_random == null)
+                {
+                    s_random = new RNGCryptoServiceProvider();
+                }
+                return s_random;
+            }
+        }
+
+        internal static HashAlgorithm CreateHashAlgorithm(string algorithm)
+        {
+            object algorithmObject = GetAlgorithmFromConfig(algorithm);
+
+            if (algorithmObject != null)
+            {
+                HashAlgorithm hashAlgorithm = algorithmObject as HashAlgorithm;
+                if (hashAlgorithm != null)
+                {
+                    return hashAlgorithm;
+                }
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new InvalidOperationException(SR.Format(SR.CustomCryptoAlgorithmIsNotValidHashAlgorithm, algorithm)));
+            }
+
+            switch (algorithm)
+            {
+                case SHAString:
+                case SHA1String:
+                case SystemSecurityCryptographySha1String:
+                case SecurityAlgorithms.Sha1Digest:
+#pragma warning disable CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                    return SHA1.Create();
+#pragma warning restore CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                case SHA256String:
+                case SecurityAlgorithms.Sha256Digest:
+                    return SHA256.Create();
+                default:
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new InvalidOperationException(SR.Format(SR.UnsupportedCryptoAlgorithm, algorithm)));
+            }
+        }
+
+        private static object GetDefaultAlgorithm(string algorithm)
+        {
+            if (string.IsNullOrEmpty(algorithm))
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentNullException("algorithm"));
+            }
+
+            switch (algorithm)
+            {
+                //case SecurityAlgorithms.RsaSha1Signature:
+                //case SecurityAlgorithms.DsaSha1Signature:
```

Comment:
These two should be resolvable.  http://www.w3.org/2000/09/xmldsig#rsa-sha1 => System.Security.Cryptography.Xml.RSAPKCS1SHA1SignatureDescription; http://www.w3.org/2000/09/xmldsig#dsa-sha1 => System.Security.Cryptography.Xml.DSASignatureDescription

Reply:
We can't new either of those classes as they are internal so we can't return them from this method. They will be created by our call to CryptoConfig elsewhere so we still support them, just not via this code path.

---
In reply to: [274000137](https://github.com/dotnet/wcf/pull/3538#discussion_r274000137) [](ancestors = 274000137)